utils/post_utils.jsx

```diff
@@ -129,3 +129,38 @@ export function containsAtChannel(text) {
 
     return (/\B@(all|channel)\b/i).test(mentionableText);
 }
+
+export function shouldFocusMainTextbox(e, activeElement) {
+    if (!e) {
+        return false;
+    }
+
+    // Do not focus if we're currently focused on a textarea or input
+    const keepFocusTags = ['TEXTAREA', 'INPUT'];
+    if (!activeElement || keepFocusTags.includes(activeElement.tagName)) {
+        return false;
+    }
+
+    // Focus if it is an attempted paste
+    if (Utils.cmdOrCtrlPressed(e) && Utils.isKeyPressed(e, Constants.KeyCodes.V)) {
+        return true;
+    }
+
+    // Do not focus if a modifier key is pressed
+    if (e.ctrlKey || e.metaKey || e.altKey) {
+        return false;
+    }
+
+    // Do not focus if the key is undefined or null
+    if (e.key == null) {
+        return false;
+    }
+
+    // Do not focus for non-character or non-number keys
+    // Note this does not work for some non-English characters
+    if (e.key.length !== 1 || !e.key.match(/[a-zA-ZÀ-ÿ0-9!@#$%^&*()-=+[\]{}\\|;:'",.<>/?`~ ]/i)) {
```

Comment:
Instead of the complicated Regex pattern, what if you just did `e.key.match(/./)` so that it matches any printable character?

Reply:
Actually, is that check even necessary? Are there any special keys that we don't want to capture that have a length of 1 for their `e.key` value?

Reply:
Actually I'd rather leave the check in and be more restrictive to start. Plus I'd also need to check for SPACE as that needs to pass